regression: aws-asg: ASG not scaling in after drain

I don't believe this is due to configuration since we have not had any major changes, but I will try an older release that was known to work.   

The autoscaled decides to scale in, marks nodes ineligible, drains them and it completes, but then fails to acknowledge the action and the ASG never scales in.  

This is a build from master @ `55d8d62`

Here is the log (trimmed consul-template output and `queue=horizontal` entries)
[autoscaler.err.log](https://github.com/hashicorp/nomad-autoscaler/files/5998308/autoscaler.err.log)

An example of the policy configuration is: 
```
  2 scaling "applications" {
  3   enabled = true
  4   min     = 7
  5   max     = 15
  6
  7   policy {
  8     cooldown            = "10m"
  9     evaluation_interval = "7m"
 10
 11     check "applications-scale-cpu-utilization" {
 12       source = "datadog"
 13       query_window = "5m"
 14       query = "sum:nomad.client.allocated.cpu{region:us-east-1, class:applications, environment:staging }*100/(sum:nomad.client.allocated.cpu{region:us-east-1, class:applications, environment:staging }+sum:nomad.client.unallocated.cpu{region:us-east-1, class:applications, environment:staging })"
 15       strategy "target-value" {
 16         # Use defined boundaries to compute fractional threshold
 17         target = 77.5
 18         threshold = 0.16129032258064516
 19       }
 20     }
 21
 22     check "applications-scale-memory-utilization" {
 23       source = "datadog"
 24       query_window = "5m"
 25       query = "sum:nomad.client.allocated.memory{region:us-east-1, class:applications, environment:staging }*100/(sum:nomad.client.allocated.memory{region:us-east-1, class:applications, environment:staging }+sum:nomad.client.unallocated.memory{region:us-east-1, class:applications, environment:staging })"
 26       strategy "target-value" {
 27         # Use defined boundaries to compute fractional threshold
 28         target = 77.5
 29         threshold = 0.16129032258064516
 30       }
 31     }
 32
 33     check "applications-scale-blocked-evaluations" {
 34       source = "datadog"
 35       query_window = "5m"
 36       query = "default_zero(default_zero(sum:nomad.aggregator.class.blocked_evaluations{region:us-east-1, class:applications, environment:staging }) / default_zero(sum:nomad.aggregator.class.blocked_evaluations{region:us-east-1, class:applications, environment:staging }))"
 37       strategy "target-value" {
 38         target = 0.9
 39       }
 40     }
 41
 42     target "aws-asg" {
 43       aws_asg_name        = "an-asg-group-name-was-here"
 44       node_class          = "applications"
 45       node_drain_deadline = "5m"
 46       node_drain_ignore_system_jobs = "true"
 47     }
 48   }
 49 }
 50
 ```

```
/ $ cat /secrets/config.hcl
nomad {
  address = "http://10.2.71.1:4646"
  region  = "staging"
}

policy_eval {
  ack_timeout    = "10m"
  delivery_limit = 4

  workers = {
    cluster    = 20
    horizontal = 20
  }
}

apm "datadog" {
  driver = "datadog"
  config = {
    dd_api_key = "..."
    dd_app_key = "..."
  }
}

target "aws-asg" {
  driver = "aws-asg"
  config = {
    aws_region = "us-east-1"
  }
}
strategy "target-value" {
  driver = "target-value"
}

telemetry {
  dogstatsd_address = "172.17.0.1:8125"
}
```

Please let me know if I can provide any more details

Thanks for the update @baxor, and thanks for trying an unrelease version, getting early feedback is incrdibly helpful to us.

After inspecting the logs there seems to be an incorrect call to check the target status after the nodes are drained. 

Filtering the logs to focus on a specific policy, we have these messages:

```
2021-02-17T18:58:53.305Z [DEBUG] file_policy_source: starting file policy monitor: file=/alloc/policies/node-class-scaling-policies.hcl name=high-memory-applications policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f
2021-02-17T19:05:53.779Z [DEBUG] policy_eval.broker: enqueue eval: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster token=
2021-02-17T19:05:53.779Z [DEBUG] policy_eval.broker: eval enqueued: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster token=
2021-02-17T19:05:53.779Z [DEBUG] policy_eval.broker: eval dequeued: queue=cluster eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f token=4146c1fd-9666-2908-a476-7f6feaf36a5b
2021-02-17T19:05:53.779Z [DEBUG] policy_eval.worker: received policy for evaluation: id=86c7397b-59a8-57f8-51ec-adf258bd8253 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster target=aws-asg
2021-02-17T19:05:53.779Z [DEBUG] policy_eval.worker: fetching current count: id=86c7397b-59a8-57f8-51ec-adf258bd8253 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster target=aws-asg
2021-02-17T19:05:54.054Z [DEBUG] policy_eval.worker.check_handler: received policy check for evaluation: check=high-memory-applications-scale-cpu-utilization id=86c7397b-59a8-57f8-51ec-adf258bd8253 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster source=datadog strategy=target-value target=aws-asg
2021-02-17T19:05:54.055Z [DEBUG] policy_eval.worker.check_handler: querying source: check=high-memory-applications-scale-cpu-utilization id=86c7397b-59a8-57f8-51ec-adf258bd8253 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster source=datadog strategy=target-value target=aws-asg query="sum:nomad.client.allocated.cpu{region:us-east-1, class:high-memory-applications, environment:staging }*100/(sum:nomad.client.allocated.cpu{region:us-east-1, class:high-memory-applications, environment:staging }+sum:nomad.client.unallocated.cpu{region:us-east-1, class:high-memory-applications, environment:staging })" source=datadog
2021-02-17T19:05:54.149Z [DEBUG] policy_eval.worker.check_handler: calculating new count: check=high-memory-applications-scale-cpu-utilization id=86c7397b-59a8-57f8-51ec-adf258bd8253 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster source=datadog strategy=target-value target=aws-asg count=5
2021-02-17T19:05:54.149Z [DEBUG] policy_eval.worker.check_handler: received policy check for evaluation: check=high-memory-applications-scale-memory-utilization id=86c7397b-59a8-57f8-51ec-adf258bd8253 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster source=datadog strategy=target-value target=aws-asg
2021-02-17T19:05:54.149Z [DEBUG] policy_eval.worker.check_handler: querying source: check=high-memory-applications-scale-memory-utilization id=86c7397b-59a8-57f8-51ec-adf258bd8253 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster source=datadog strategy=target-value target=aws-asg query="sum:nomad.client.allocated.memory{region:us-east-1, class:high-memory-applications, environment:staging }*100/(sum:nomad.client.allocated.memory{region:us-east-1, class:high-memory-applications, environment:staging }+sum:nomad.client.unallocated.memory{region:us-east-1, class:high-memory-applications, environment:staging })" source=datadog
2021-02-17T19:05:54.256Z [DEBUG] policy_eval.worker.check_handler: calculating new count: check=high-memory-applications-scale-memory-utilization id=86c7397b-59a8-57f8-51ec-adf258bd8253 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster source=datadog strategy=target-value target=aws-asg count=5
2021-02-17T19:05:54.256Z [DEBUG] policy_eval.worker.check_handler: received policy check for evaluation: check=high-memory-applications-scale-blocked-evaluations id=86c7397b-59a8-57f8-51ec-adf258bd8253 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster source=datadog strategy=target-value target=aws-asg
2021-02-17T19:05:54.256Z [DEBUG] policy_eval.worker.check_handler: querying source: check=high-memory-applications-scale-blocked-evaluations id=86c7397b-59a8-57f8-51ec-adf258bd8253 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster source=datadog strategy=target-value target=aws-asg query="default_zero(default_zero(sum:nomad.aggregator.class.blocked_evaluations{region:us-east-1, class:high-memory-applications, environment:staging }) / default_zero(sum:nomad.aggregator.class.blocked_evaluations{region:us-east-1, class:high-memory-applications, environment:staging }))" source=datadog
2021-02-17T19:05:54.342Z [DEBUG] policy_eval.worker.check_handler: calculating new count: check=high-memory-applications-scale-blocked-evaluations id=86c7397b-59a8-57f8-51ec-adf258bd8253 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster source=datadog strategy=target-value target=aws-asg count=5
2021-02-17T19:05:54.343Z [INFO]  policy_eval.worker: scaling target: id=86c7397b-59a8-57f8-51ec-adf258bd8253 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster target=aws-asg from=5 to=3 reason="scaling down because factor is 0.544280" meta=map[nomad_policy_id:15901f2c-27bd-5a1b-70f3-f13039c2706f]
2021-02-17T19:05:54.469Z [DEBUG] internal_plugin.aws-asg: filtering node list: filter=class value=high-memory-applications
2021-02-17T19:05:54.469Z [DEBUG] internal_plugin.aws-asg: identified Nomad node for removal: node_id=2cd8c564-7c54-f22a-7457-e1322a8c956b
2021-02-17T19:05:54.469Z [DEBUG] internal_plugin.aws-asg: identified Nomad node for removal: node_id=ce10694c-cc3c-4a66-6d5c-0b946fd4f4c3
2021-02-17T19:05:54.472Z [DEBUG] internal_plugin.aws-asg: identified remote provider ID for node: node_id=2cd8c564-7c54-f22a-7457-e1322a8c956b remote_id=i-045b409b620d02d9a
2021-02-17T19:05:54.474Z [DEBUG] internal_plugin.aws-asg: identified remote provider ID for node: node_id=ce10694c-cc3c-4a66-6d5c-0b946fd4f4c3 remote_id=i-00b6aa87f4fb87f2f
2021-02-17T19:05:54.474Z [INFO]  internal_plugin.aws-asg: triggering drain on node: node_id=ce10694c-cc3c-4a66-6d5c-0b946fd4f4c3 deadline=5m0s
2021-02-17T19:05:54.474Z [INFO]  internal_plugin.aws-asg: triggering drain on node: node_id=2cd8c564-7c54-f22a-7457-e1322a8c956b deadline=5m0s
2021-02-17T19:05:55.617Z [DEBUG] internal_plugin.aws-asg: received node drain message: node_id=2cd8c564-7c54-f22a-7457-e1322a8c956b msg="Alloc "82333427-eea6-dfaa-ff02-1306fe298ca1" marked for migration"
2021-02-17T19:05:55.617Z [DEBUG] internal_plugin.aws-asg: received node drain message: node_id=2cd8c564-7c54-f22a-7457-e1322a8c956b msg="Alloc "010f9b4a-183b-1f53-426a-c0d4ccfc002b" marked for migration"
2021-02-17T19:05:55.857Z [DEBUG] internal_plugin.aws-asg: received node drain message: node_id=2cd8c564-7c54-f22a-7457-e1322a8c956b msg="Alloc "82333427-eea6-dfaa-ff02-1306fe298ca1" draining"
2021-02-17T19:05:55.885Z [DEBUG] internal_plugin.aws-asg: received node drain message: node_id=2cd8c564-7c54-f22a-7457-e1322a8c956b msg="Alloc "010f9b4a-183b-1f53-426a-c0d4ccfc002b" draining"
2021-02-17T19:06:03.371Z [DEBUG] internal_plugin.aws-asg: received node drain message: node_id=2cd8c564-7c54-f22a-7457-e1322a8c956b msg="Alloc "82333427-eea6-dfaa-ff02-1306fe298ca1" status running -> complete"
2021-02-17T19:06:04.611Z [DEBUG] internal_plugin.aws-asg: received node drain message: node_id=2cd8c564-7c54-f22a-7457-e1322a8c956b msg="Alloc "010f9b4a-183b-1f53-426a-c0d4ccfc002b" status running -> complete"
2021-02-17T19:06:20.614Z [DEBUG] internal_plugin.aws-asg: received node drain message: node_id=ce10694c-cc3c-4a66-6d5c-0b946fd4f4c3 msg="Alloc "8c5247ae-4979-7a2a-07a4-20e69fae1d4a" marked for migration"
2021-02-17T19:06:20.869Z [INFO]  internal_plugin.aws-asg: received node drain message: node_id=ce10694c-cc3c-4a66-6d5c-0b946fd4f4c3 msg="Drain complete for node ce10694c-cc3c-4a66-6d5c-0b946fd4f4c3"
2021-02-17T19:06:20.879Z [DEBUG] internal_plugin.aws-asg: received node drain message: node_id=ce10694c-cc3c-4a66-6d5c-0b946fd4f4c3 msg="Alloc "8c5247ae-4979-7a2a-07a4-20e69fae1d4a" draining"
2021-02-17T19:06:27.819Z [DEBUG] internal_plugin.aws-asg: received node drain message: node_id=ce10694c-cc3c-4a66-6d5c-0b946fd4f4c3 msg="Alloc "8c5247ae-4979-7a2a-07a4-20e69fae1d4a" status running -> complete"
2021-02-17T19:06:45.022Z [DEBUG] internal_plugin.aws-asg: received node drain message: node_id=2cd8c564-7c54-f22a-7457-e1322a8c956b msg="Alloc "d215738d-0035-641a-25bd-d9cd60c4d951" marked for migration"
2021-02-17T19:06:45.294Z [DEBUG] internal_plugin.aws-asg: received node drain message: node_id=2cd8c564-7c54-f22a-7457-e1322a8c956b msg="Alloc "d215738d-0035-641a-25bd-d9cd60c4d951" draining"
2021-02-17T19:06:45.348Z [INFO]  internal_plugin.aws-asg: received node drain message: node_id=2cd8c564-7c54-f22a-7457-e1322a8c956b msg="Drain complete for node 2cd8c564-7c54-f22a-7457-e1322a8c956b"
2021-02-17T19:06:51.573Z [DEBUG] internal_plugin.aws-asg: received node drain message: node_id=2cd8c564-7c54-f22a-7457-e1322a8c956b msg="Alloc "d215738d-0035-641a-25bd-d9cd60c4d951" status running -> complete"
2021-02-17T19:12:53.325Z [WARN]  internal_plugin.aws-asg: node pool status readiness check failed: error="node 2cd8c564-7c54-f22a-7457-e1322a8c956b is ineligible, node ce10694c-cc3c-4a66-6d5c-0b946fd4f4c3 is ineligible"
2021-02-17T19:15:53.780Z [DEBUG] policy_eval.broker: nack eval: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 token=4146c1fd-9666-2908-a476-7f6feaf36a5b
2021-02-17T19:15:53.780Z [DEBUG] policy_eval.broker: enqueue eval: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster token=4146c1fd-9666-2908-a476-7f6feaf36a5b
2021-02-17T19:15:53.780Z [DEBUG] policy_eval.broker: eval enqueued: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster token=4146c1fd-9666-2908-a476-7f6feaf36a5b
2021-02-17T19:15:53.780Z [INFO]  policy_eval.broker: eval nack'd, retrying it: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f token=4146c1fd-9666-2908-a476-7f6feaf36a5b
2021-02-17T19:15:53.780Z [DEBUG] policy_eval.broker: eval dequeued: queue=cluster eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f token=54b19730-1889-e76e-c127-497f5a7cfeb4
2021-02-17T19:15:53.780Z [DEBUG] policy_eval.worker: received policy for evaluation: id=ca5ef52d-37ff-090a-9176-79ec925113da policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster target=aws-asg
2021-02-17T19:15:53.780Z [DEBUG] policy_eval.worker: fetching current count: id=ca5ef52d-37ff-090a-9176-79ec925113da policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster target=aws-asg
2021-02-17T19:15:53.794Z [WARN]  internal_plugin.aws-asg: node pool status readiness check failed: error="node 2cd8c564-7c54-f22a-7457-e1322a8c956b is ineligible, node ce10694c-cc3c-4a66-6d5c-0b946fd4f4c3 is ineligible"
2021-02-17T19:15:53.794Z [ERROR] policy_eval.worker: failed to evaluate policy: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 eval_token=54b19730-1889-e76e-c127-497f5a7cfeb4 id=ca5ef52d-37ff-090a-9176-79ec925113da policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster err="target not ready"
2021-02-17T19:15:53.794Z [DEBUG] policy_eval.broker: nack eval: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 token=54b19730-1889-e76e-c127-497f5a7cfeb4
2021-02-17T19:15:53.794Z [DEBUG] policy_eval.broker: enqueue eval: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster token=54b19730-1889-e76e-c127-497f5a7cfeb4
2021-02-17T19:15:53.794Z [DEBUG] policy_eval.broker: eval enqueued: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster token=54b19730-1889-e76e-c127-497f5a7cfeb4
2021-02-17T19:15:53.794Z [INFO]  policy_eval.broker: eval nack'd, retrying it: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f token=54b19730-1889-e76e-c127-497f5a7cfeb4
2021-02-17T19:15:53.794Z [DEBUG] policy_eval.broker: dequeue eval: queue=cluster
2021-02-17T19:15:53.794Z [DEBUG] policy_eval.broker: eval dequeued: queue=cluster eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f token=33ca51a2-4116-e5ea-ef4f-bc386f0612eb
2021-02-17T19:15:53.794Z [DEBUG] policy_eval.worker: received policy for evaluation: id=4b5071af-b763-8c1d-9a93-1407b9e66f69 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster target=aws-asg
2021-02-17T19:15:53.794Z [DEBUG] policy_eval.worker: fetching current count: id=4b5071af-b763-8c1d-9a93-1407b9e66f69 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster target=aws-asg
2021-02-17T19:15:53.794Z [DEBUG] policy_eval.broker: waiting for eval: queue=cluster
2021-02-17T19:15:53.808Z [WARN]  internal_plugin.aws-asg: node pool status readiness check failed: error="node 2cd8c564-7c54-f22a-7457-e1322a8c956b is ineligible, node ce10694c-cc3c-4a66-6d5c-0b946fd4f4c3 is ineligible"
2021-02-17T19:15:53.808Z [ERROR] policy_eval.worker: failed to evaluate policy: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 eval_token=33ca51a2-4116-e5ea-ef4f-bc386f0612eb id=4b5071af-b763-8c1d-9a93-1407b9e66f69 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster err="target not ready"
2021-02-17T19:15:53.808Z [DEBUG] policy_eval.broker: nack eval: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 token=33ca51a2-4116-e5ea-ef4f-bc386f0612eb
2021-02-17T19:15:53.808Z [DEBUG] policy_eval.broker: enqueue eval: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster token=33ca51a2-4116-e5ea-ef4f-bc386f0612eb
2021-02-17T19:15:53.808Z [DEBUG] policy_eval.broker: eval enqueued: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster token=33ca51a2-4116-e5ea-ef4f-bc386f0612eb
2021-02-17T19:15:53.808Z [INFO]  policy_eval.broker: eval nack'd, retrying it: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f token=33ca51a2-4116-e5ea-ef4f-bc386f0612eb
2021-02-17T19:15:53.808Z [DEBUG] policy_eval.broker: dequeue eval: queue=cluster
2021-02-17T19:15:53.808Z [DEBUG] policy_eval.broker: eval dequeued: queue=cluster eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f token=b8235eca-aa1a-b91a-00b9-c4195d8de0c3
2021-02-17T19:15:53.808Z [DEBUG] policy_eval.worker: received policy for evaluation: id=4b5071af-b763-8c1d-9a93-1407b9e66f69 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster target=aws-asg
2021-02-17T19:15:53.808Z [DEBUG] policy_eval.worker: fetching current count: id=4b5071af-b763-8c1d-9a93-1407b9e66f69 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster target=aws-asg
2021-02-17T19:15:53.808Z [DEBUG] policy_eval.broker: waiting for eval: queue=cluster
2021-02-17T19:15:53.823Z [WARN]  internal_plugin.aws-asg: node pool status readiness check failed: error="node 2cd8c564-7c54-f22a-7457-e1322a8c956b is ineligible, node ce10694c-cc3c-4a66-6d5c-0b946fd4f4c3 is ineligible"
2021-02-17T19:15:53.823Z [ERROR] policy_eval.worker: failed to evaluate policy: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 eval_token=b8235eca-aa1a-b91a-00b9-c4195d8de0c3 id=4b5071af-b763-8c1d-9a93-1407b9e66f69 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f queue=cluster err="target not ready"
2021-02-17T19:15:53.823Z [DEBUG] policy_eval.broker: nack eval: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 token=b8235eca-aa1a-b91a-00b9-c4195d8de0c3
2021-02-17T19:15:53.823Z [WARN]  policy_eval.broker: eval delivery limit reached: eval_id=e7b51423-fe34-ee03-1b3b-240324b736f4 policy_id=15901f2c-27bd-5a1b-70f3-f13039c2706f token=b8235eca-aa1a-b91a-00b9-c4195d8de0c3 count=4 limit=4
```

The troublesome call is this one:

```
2021-02-17T19:12:53.325Z [WARN]  internal_plugin.aws-asg: node pool status readiness check failed: error="node 2cd8c564-7c54-f22a-7457-e1322a8c956b is ineligible, node ce10694c-cc3c-4a66-6d5c-0b946fd4f4c3 is ineligible"
```

From this point on, the policy evaluation will always fail because the cluster is left in a unfinished state 😕 

I'm not sure why the plugin is trying to check the cluster status after draining the nodes, it should proceed with detaching the nodes from the ASG, but [this log message](https://github.com/hashicorp/nomad-autoscaler/blob/master/plugins/builtin/target/aws-asg/plugin/aws.go#L125-L126) is never emitted.

We are investigating this further and we'll keep you updated.

--- a/sdk/helper/scaleutils/nomad.go
+++ b/sdk/helper/scaleutils/nomad.go
@@ -293,7 +293,7 @@ func (si *ScaleIn) drainNode(ctx context.Context, nodeID string, spec *api.Drain
 
 	// Monitor the drain so we output the log messages. An error here indicates
 	// the drain failed to complete successfully.
-	if err := si.monitorNodeDrain(ctx, nodeID, resp.LastIndex); err != nil {
+	if err := si.monitorNodeDrain(ctx, nodeID, resp.LastIndex, spec.IgnoreSystemJobs); err != nil {
 		return fmt.Errorf("context done while monitoring node drain: %v", err)
 	}
 	return nil
@@ -301,11 +301,8 @@ func (si *ScaleIn) drainNode(ctx context.Context, nodeID string, spec *api.Drain
 
 // monitorNodeDrain follows the drain of a node, logging the messages we
 // receive to their appropriate level.
-//
-// TODO(jrasell): currently the ignoreSys param is hardcoded to false, we will
-//  probably want to expose this to operators in the future.
-func (si *ScaleIn) monitorNodeDrain(ctx context.Context, nodeID string, index uint64) error {
-	for msg := range si.nomad.Nodes().MonitorDrain(ctx, nodeID, index, false) {
+func (si *ScaleIn) monitorNodeDrain(ctx context.Context, nodeID string, index uint64, ignoreSys bool) error {
+	for msg := range si.nomad.Nodes().MonitorDrain(ctx, nodeID, index, ignoreSys) {
 		switch msg.Level {
 		case api.MonitorMsgLevelInfo:
 			si.log.Info("received node drain message", "node_id", nodeID, "msg", msg.Message)

